### DISCUSSION TITLE
Cats are infested

### DISCUSSION THREAD

**Discussion by M1ngM1ngQQY** [ultralytics/ultralytics#7210] [Ideas]
I would like to ask, let's say my model can recognize these three categories (people, cats, dogs), I use a camera to monitor at my door, I detect a cat appearing, how do I get this information. I want to make a cat that appears, and then my UI pops up with "cat infested", how should I do it?

**Comment by pderrenger** [comment]
@M1ngM1ngQQY to achieve this, you can use the YOLOv8 model in predict mode to detect objects in the video stream from your camera. Once the model identifies a cat, you can send a signal to your UI to display the "cat infested" message. Here's a simplified example of how you might set this up in Python:

```python
from ultralytics import YOLO

# Load your trained model
model = YOLO('path/to/your/model.pt')

# Predict in real-time from your camera (assuming camera index is 0)
results = model.predict(source=0, stream=True)

for result in results:
    # Check if 'cat' is detected in the prediction
    for label, confidence, bbox in zip(result.labels, result.confidences, result.boxes):
        if label == 'cat':
            # Code to update your UI with "cat infested"
            update_ui_with_cat_alert()
            break  # Stop after the first cat is found
```

In this example, `update_ui_with_cat_alert()` is a placeholder for the function you would implement to update your UI. The `predict` method is set to stream mode, which means it will process frames as they come in real-time. When a cat is detected, it triggers the UI update function.

Remember to replace `'path/to/your/model.pt'` with the actual path to your trained model file. Also, ensure that your UI update function is designed to handle the alert appropriately.
